cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -3454,8 +3454,8 @@ void RegisterHashAggregateBasic(FunctionRegistry* registry) {
   }
 
   {
-    auto func = std::make_shared<HashAggregateFunction>("hash_count_all", Arity::Unary(),
-                                                        hash_count_all_doc, NULLPTR);
+    auto func = std::make_shared<HashAggregateFunction>(
+        "hash_count_all", Arity::Nullary(), hash_count_all_doc, NULLPTR);
 
     DCHECK_OK(func->AddKernel(MakeUnaryKernel(HashAggregateInit<GroupedCountAllImpl>)));
```

Comment:
Maybe remove `MakeUnaryKernel` and  add a `MakeNullaryKernel`?

Reply:
I'm actually undoing this code change and keeping only the documentation change. All aggregation functions are documented to have one less arity than what is in the documents (eg: `hash_count_distinct` is declared `Binary` in the code, but users only have to pass one argument, so it's documented as `Unary`).

`hash_count_all` is correctly configured as `Arity::Unary()`, but users pass unary-1 arguments, so it should be `Nullary` in the documentation. 

(I worked on n-ary (0 and 1+) aggregate functions in #15083 more than a year ago)

Reply:
I had forgotten this: all hash aggregate kernels receive an extra argument containing group ids. That's an argument to the kernel but *not* to the Function (because group IDs must be assembled from the batch)